base/replutil.jl

```diff
@@ -147,11 +147,13 @@ function showerror(io::IO, ex::BoundsError)
         print(io, ": attempt to access ")
         if isa(ex.a, AbstractArray)
             print(io, summary(ex.a))
+        elseif isa(ex.a, Tuple) && isdefined(ex, :i) && isa(ex.i, Tuple)
```

Comment:
I don't like the ambiguity either. Some options I see:
- create a new exception type (yuck)
- add another field to `BoundsError`
- create an internal `ArrayIndices` type that wraps the tuple-of-indices and sticks it in the `a` slot

Of these, I like the third the best.